tests/stub/optimizations/test_optimizations.py

```diff
@@ -55,6 +59,51 @@ def test():
                     test()
                 self._server.reset()
 
+    @driver_feature(types.Feature.API_DRIVER_EXECUTE_QUERY,
+                    types.Feature.OPT_EXECUTE_QUERY_PIPELINING,
+                    types.Feature.BOLT_5_3)
+    def test_execute_query_pipelines_begin(self):
+        script = "begin_pipeline.script"
+        self._server.start(path=self.script_path("v5x3", script))
+
+        auth = types.AuthorizationToken("basic", principal="neo4j",
+                                        credentials="pass")
+        driver = Driver(self._backend, f"bolt://{self._server.address}", auth)
+
+        driver.execute_query("RETURN 1 AS n")
+
+        driver.close()
+        self._server.done()
+
+    @driver_feature(types.Feature.API_DRIVER_EXECUTE_QUERY,
+                    types.Feature.OPT_EXECUTE_QUERY_PIPELINING,
+                    types.Feature.BOLT_5_3)
+    def test_execute_query_does_not_pipeline_commit(self):
+        script = "commit_pipeline.script"
```

Comment:
This test seems wrong to me, we are verifying that bolt stub shuts down when sent a message in the wrong order.

Reply:
this is a case that is implicitly tested, by the above test

Reply:
this is the only reliable means of testing this behaviour(currently) so I withdraw this comment.